capi/geos_ts_c.cpp

```diff
@@ -2752,7 +2747,7 @@ extern "C" {
             GEOSContextHandleInternal_t* handle = reinterpret_cast<GEOSContextHandleInternal_t*>(extHandle);
             const GeometryFactory* gf = handle->geomFactory;
 
-            return gf->createPoint(cs);
+            return gf->createPoint(std::unique_ptr<CoordinateSequence>(cs)).release();
```

Comment:
Ewwww?

Reply:
```
#define 🪂 .release()
#define 👉 std::unique_ptr
#define 🧬 CoordinateSequence

return gf->createPoint(👉<🧬>(cs)).🪂
```

Reply:
That is much prettier. However, my implicit question is "why"? It seems completely self-referential to wrap the raw pointer and then release it in one go.

Reply:
Because the C API only deals in raw pointers, so we have to either (a) create and release them at this boundary between C and C++ or (b) maintain a separate set of `GeometryFactory` methods dealing in raw pointers and hope they're not used within the C++ lib itself. This one line is ugly but I think the overall change is strongly net-positive.

Reply:
My apologies, I didn't interpret the `))` correctly. I thought you were doing
```
std::unique_ptr<CoordinateSequence>(cs).release()
```
I see now that the release is a (very important) `)` later.